---
source: crates/analyzer/tests/errors.rs
expression: "error_string(&path, test_files::fixture(path))"
---
error: the function `do_private_thingz` on `struct Foo` is private
  ┌─ compile_errors/call_non_pub_fn_on_struct2.fe:8:15
  │
2 │     fn do_private_thingz(self) {
  │        ----------------- `do_private_thingz` is defined here
  ·
8 │         Foo().do_private_thingz()
  │               ^^^^^^^^^^^^^^^^^ this function is not `pub`
  │
  = `do_private_thingz` can only be called from other functions within `Foo`
  = Hint: use `pub fn do_private_thingz(..)` to make `do_private_thingz` callable from outside of `Foo`


